### PR TITLE
[EBPF-577] gpu: add e2e test to check presence of metrics

### PR DESCRIPTION
**Pull request by gjulianm**
### What does this PR do?

This PR adds a e2e test that ensures we have metrics received in the fakeintake

### CODE REVIEW

File: test/new-e2e/tests/gpu/gpu_test.go [hunk 1]
```diff
@@ -104,3 +108,20 @@ func (v *gpuSuite) TestGPUCheckIsEnabled() {
...
+
+func (v *gpuSuite) TestVectorAddProgramDetected() {
+	vm := v.Env().RemoteHost
+
+	out, err := vm.Execute(fmt.Sprintf("docker run --rm --gpus all %s", vectorAddDockerImg))
+	v.Require().NoError(err)
+	v.Require().NotEmpty(out)
+
+	v.EventuallyWithT(func(c *assert.CollectT) {
+		metricNames := []string{"gpu.memory", "gpu.utilization", "gpu.max_memory"}
```
Comment (val06) [inline test/new-e2e/tests/gpu/gpu_test.go:120]: didn't we call it `gpu.memory.max`?

**Review by val06 — COMMENTED** [review]
reviewed

File: test/new-e2e/tests/gpu/gpu_test.go (hunk 1 above)
Comment (gjulianm) [reply test/new-e2e/tests/gpu/gpu_test.go:120]: We did, fixed

**Review by val06 — APPROVED** [review]

**Pull request merged**